From 4df1827044d3e8181aa52fc45c184498f16b30da Mon Sep 17 00:00:00 2001
From: Manjunath <manjunathx.b.s@intel.com>
Date: Tue, 14 Aug 2012 10:53:22 +0530
Subject: [PORT FROM MAIN] [PATCH] JB: EAS_Prepare failed: [-17]. when playing mxmf file

BZ: 51232

This patch contains changes in eas_mdls.c. In DLSParser, the
file with artCount ( no. of articulations ) = 0 is considered
valid.

Change-Id: I27c2198da11a0b3b60b751ff04c092e25147d623
Orig-Change-Id: I59c67bc778178d1c49f00fff018068f0ac15ab5c
Signed-off-by: Manjunath <manjunathx.b.s@intel.com>
Reviewed-on: http://android.intel.com:8080/82125
Reviewed-by: M, Arulselvan <arulselvan.m@intel.com>
Tested-by: M, Arulselvan <arulselvan.m@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 arm-wt-22k/lib_src/eas_mdls.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/arm-wt-22k/lib_src/eas_mdls.c b/arm-wt-22k/lib_src/eas_mdls.c
index e8dc463..01a7f98 100644
--- a/arm-wt-22k/lib_src/eas_mdls.c
+++ b/arm-wt-22k/lib_src/eas_mdls.c
@@ -596,7 +596,8 @@ EAS_RESULT DLSParser (EAS_HW_DATA_HANDLE hwInstData, EAS_FILE_HANDLE fileHandle,
         }
 
         /* limit check  */
-        if ((dls.artCount == 0) || (dls.artCount > DLS_MAX_ART_COUNT))
+        //if ((dls.artCount == 0) || (dls.artCount > DLS_MAX_ART_COUNT))
+        if (dls.artCount > DLS_MAX_ART_COUNT) //Fix for BZ-51232. Considering file with artCount = 0 as valid DLS file.
         {
             { /* dpp: EAS_ReportEx(_EAS_SEVERITY_ERROR, "DLS file contains invalid #articulations [%u]\n", dls.regionCount); */ }
             return EAS_ERROR_FILE_FORMAT;
-- 
1.7.4.1

